neutrinordp/xrdp-neutrinordp.c

```diff
@@ -609,6 +610,77 @@ lxrdp_set_param(struct mod *mod, const char *name, const char *value)
         {
             mod->perf_settings_values_mask |= PERF_DISABLE_CURSOR_SHADOW;
         }
+    }
+        else if (g_strcmp(name, "neutrinordp.allow_client_keyboardLayout") == 0)
+    {
+        if (g_text2bool(value))
+        {
+            if (settings->kbd_type == 0)
+            {
+                settings->kbd_type = mod->client_info.keyboard_type;
+            }
+            if (settings->kbd_subtype == 0)
+            {
+                settings->kbd_subtype = mod->client_info.keyboard_subtype;
+            }
+            if (settings->kbd_fn_keys == 0)
+            {
+                /* Define the most common number of function keys, 12. 
+                   because we can't get it from client. */
+                settings->kbd_fn_keys = 12;
+            }
+            if (settings->kbd_layout == 0)
+            {
+                settings->kbd_layout = mod->client_info.keylayout;
+                /* When the keyboard type is 0x07 (Japanese keyboard), 
+                   this function masks the upper 16 bits.*/
+                if (mod->client_info.keyboard_type == 0x07)
+                {
+                    LOG(LOG_LEVEL_DEBUG,
+                    "keyboardLayout mask for Japanese PC106/109 keyboard:"
+                    "[0x%08X]", 0x0000FFFF);
+                    settings->kbd_layout &= 0x0000FFFF;
+                }
+            }
+            LOG(LOG_LEVEL_INFO, "NeutrinoRDP proxy remote keyboard settings, "
+                                "kbd_type:[0x%02X], kbd_subtype:[0x%02X], "
+                                "kbd_fn_keys:[%02d], kbd_layout:[0x%08X]",
+                                settings->kbd_type, settings->kbd_subtype,
+                                settings->kbd_fn_keys, settings->kbd_layout);
+        }
+    }
+    /* The following options are for DEBUG and workaround.*/
+    else if (g_strcmp(name, "neutrinordp.override_keyboardLayout_masks") == 0)
+    {
+            settings->kbd_layout =
+                          mod->client_info.keylayout & strtol(value, NULL, 16);
+            LOG(LOG_LEVEL_WARNING,
+                "%s:[%s],overrided masked kbd_layout:[0x%08X]",
+                name, value, settings->kbd_layout);
+    }
+    else if (g_strcmp(name, "neutrinordp.override_kbd_type") == 0)
+    {
+            settings->kbd_type = strtod(value, NULL);
+            LOG(LOG_LEVEL_WARNING, "%s:[%s],overrided kbd_type:[0x%02X]",
```

Comment:
Suggest 'overrode' instead of 'overrided'. Also applies to following sections.

Reply:
Thank you.
I'll ovverode it.